src/EnergyPlus/ElectricPowerServiceManager.cc

```diff
@@ -3271,6 +3397,10 @@ void ElectricStorage::reinitAtBeginEnvironment()
             }
             batteryDamage_ = 0.0;
         }
+    } else if (storageModelMode_ == StorageModelType::liIonNmcBattery) {
+        // Copy the initial battery state to the last battery state
+        *ssc_lastBatteryState_ = *ssc_initBatteryState_;
+        ssc_battery_->set_state(*ssc_lastBatteryState_);
```

Comment:
The battery model has this nifty state object that you can save and restore, which is really helpful for resetting after warmups and new environments and allowing us to run the same system timestep a few times as EnergyPlus does.